[xbd] Add support for new `AndroidAarLibrary` item

Redth opened the pull request:
Xamarin.Android in 15.7 and newer will include the `AndroidAarLibrary` item group.  .aar files added to this group will automatically be compiled into the app, so Xamarin.Build.Download no longer has to embed them into the referenced assembly that needs them as embedded resources.

Current builds of 15.7 do not include the `<AvailableItemName Include="AndroidAarLibrary" />` so this PR manually adds it to our targets for now to be able to test with.  This will need to be removed before merging.

Redth committed:
[xbd] Add support for new `AndroidAarLibrary` item

Xamarin.Android in 15.7 and newer will include the `AndroidAarLibrary` item group.  .aar files added to this group will automatically be compiled into the app, so Xamarin.Build.Download no longer has to embed them into the referenced assembly that needs them as embedded resources.

Redth commented on Util/Xamarin.Build.Download/source/Xamarin.Build.Download/Xamarin.Build.Download.targets:
```diff
@@ -107,9 +107,23 @@
 
+	<!-- TODO: AndroidAarLibrary is not yet in AvailableItemName so we added it manually for now -->
+	<ItemGroup>
+		<AvailableItemName Include="AndroidAarLibrary" />
```
This is temporary until 15.7 builds include it.

dellis1972 commented on Util/Xamarin.Build.Download/source/Xamarin.Build.Download/Xamarin.Build.Download.targets:
```diff
@@ -107,9 +107,23 @@
+		
+	<Target Name="_XamarinBuildDownloadAarInclude"
+			DependsOnTargets="_XamarinBuildDownload"
+			Condition="'@(XamarinBuildDownloadRestoreAssemblyAar)'!='' and %(AvailableItemName.Identity) == 'AndroidAarLibrary'">
```
does `%(AvailableItemName.Identity)` need to be in single quotes?

dellis1972 replied:
Thinking about it , it might be better to define a property 

```xml
<PropertyGroup>
   <SupportsAarLibrary Condition="'%(AvailableItemName.Identity)' == 'AndroidAarLibrary'">true</SupportsAarLibrary>
<SupportsAarLibrary Condition="'$(SupportsAarLibrary)' == ''">false</SupportsAarLibrary>
</PropertyGroup>
```

I know MSbuild might run a target multiple times when using `%(foo.Identity)` due to its Batching support. I'm not sure if that happens in this case. It might not, if it doesn't ignore me :)

Redth committed:
[xbd] Move aar item support to property

Also adds the include target to the master list of targets to run

Redth committed:
[xbd] fix logical test (copy paste error)

mattleibow commented on Util/Xamarin.Build.Download/source/Xamarin.Build.Download/Xamarin.Build.Download.targets:
```diff
@@ -107,9 +119,17 @@
 
+	<Target Name="_XamarinBuildDownloadAarInclude"
+			DependsOnTargets="_XamarinBuildDownload"
+			Condition="'@(XamarinBuildDownloadRestoreAssemblyAar)'!='' and '$(SupportsAarLibrary)'' == 'true'">
```
There is an extra `'` here: `'$(SupportsAarLibrary)**''** == 'true'`

mattleibow commented on Util/Xamarin.Build.Download/source/Xamarin.Build.Download/Xamarin.Build.Download.targets:
```diff
@@ -107,9 +119,17 @@
 	<Target Name="_XamarinAndroidBuildAarRestore"
 			DependsOnTargets="_XamarinBuildDownload"
-			Condition="'@(XamarinBuildDownloadRestoreAssemblyAar)'!=''"
+			Condition="'@(XamarinBuildDownloadRestoreAssemblyAar)'!='' and '$(SupportsAarLibrary)'' != 'true'"
```
Another double `''` here - that autocomplete :0

Redth committed:
[xbd] remove extra quote

Redth committed:
[xbd] Remove temp AvailableItemName declaration

Redth committed:
[xbd] preview release

Redth committed:
[xbd] Rework availableitemname checking

The property group is evaluated before the itemgroups, so we would never detect an availableitemname in the previous case.

This moves the detection straight into the target conditions.

Redth committed:
[xbd] fix proguard config target dependency

We won’t always run _XamarinAndroidBuildAarRestore, so we should instead depend on _XamarinBuildDownload to ensure the .aar files are in place before we try and process them.

Redth committed:
[xbd] Fix name, AvailableItemName not Names

newky2k committed:
Merge branch 'master' into xbd-aar-15_7

Redth committed:
Merge branch 'master' into xbd-aar-15_7

Redth committed:
Merge branch 'master' into xbd-aar-15_7

Redth commented:
Added to #696
